### PR TITLE
chore: remove priority sampler

### PR DESCRIPTION
**Pull request by emmettbutler**
This change removes the `_priority_sampler` members from `Tracer` and `AgentWriter`. `Tracer._priority_sampler` and `AgentWriter` are private, so removing these attributes doesn't break backward compatibility. The main motivation for this change is to simplify code related to sampling and make it easier to change.

The priority sampler's functionality is fully duplicated by the default `DatadogSampler`, so this change doesn't have any functional effect.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by ZStriker19 — COMMENTED** [review]
Just a couple of questions for understanding.

File: ddtrace/settings/config.py [hunk 1]
```diff
@@ -208,6 +208,7 @@ def __init__(self):
         self._trace_sample_rate = os.getenv("DD_TRACE_SAMPLE_RATE")
         self._trace_rate_limit = int(os.getenv("DD_TRACE_RATE_LIMIT", default=DEFAULT_SAMPLING_RATE_LIMIT))
         self._trace_sampling_rules = os.getenv("DD_TRACE_SAMPLING_RULES")
+        self._priority_sampling = asbool(os.getenv("DD_PRIORITY_SAMPLING", default=True))
```
[the remark this answers is not in the document]
Comment (emmettbutler) [reply ddtrace/settings/config.py:247]: `True` is the current default in `1.x`

[the remark this answers is not in the document]
Comment (ZStriker19) [reply ddtrace/settings/config.py:247]: Right, but do we just basically keep the priority sampler around like this to make sure api_version is right, or does it do anything else?

[the remark this answers is not in the document]
Comment (emmettbutler) [reply ddtrace/settings/config.py:247]: `priority_sampler.sample()` gets called in 1.x, but the tests still pass when I remove it, and it's not part of the public API, so :shrug:

**Review by majorgreys — APPROVED** [review]
Non-blocking: Can we do away with priority sampling entirely?

**Comment by emmettbutler** [comment]
@majorgreys based on a fairly recent customer issue reported about priority sampling, I suspect not. https://github.com/DataDog/dd-trace-py/commit/2bf5cacc9d25b70fa2a80dba7e531d675323c1fc

**Pull request merged**